src/workflow/mod.rs

```diff
@@ -7,15 +7,31 @@ pub(crate) use concurrency_manager::WorkflowConcurrencyManager;
 pub(crate) use driven_workflow::{ActivationListener, DrivenWorkflow, WorkflowFetcher};
 
 use crate::{
-    machines::{ProtoCommand, WFCommand, WorkflowMachines},
+    machines::{ProtoCommand, WFCommand, WFMachinesError, WorkflowMachines},
     protos::{
         coresdk::workflow_activation::WfActivation,
-        temporal::api::{history::v1::History, workflowservice::v1::PollWorkflowTaskQueueResponse},
+        temporal::api::{common::v1::WorkflowExecution, history::v1::History},
     },
-    protosext::HistoryInfo,
-    CoreError, Result,
+    protosext::{HistoryInfo, HistoryInfoError},
 };
-use std::sync::mpsc::Sender;
+use std::sync::mpsc::{SendError, Sender};
+
+type Result<T, E = WorkflowError> = std::result::Result<T, E>;
+
+/// Errors relating to workflow management and machine logic
```

Comment:
Can we mention here in the comment when this error is returned and who is expected to be on the receiving end?

Reply:
Will do. I think I can simplify more of the errors in general as well.